File: mlnba-ss/src/main/java/com/mgu/mlnba/model/Team.java
package com.mgu.mlnba.model;



import java.util.ArrayList;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection="teams")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Team {

        @Id
    protected String id;

    protected String name;
    protected String season;

    protected String description;
    
    @Builder.Default()
    @DBRef
    protected List<Member> members = new ArrayList<>();

}

File: mlnba-client/src/app/shared/team/team.ts
import { Member } from '../member/member';
import { Training } from './training';
export class Team {
    id: string;
    name: string;
    season: string;
    description: string;
    members: Member[];
    trainings: Training[] = [];
  }

export class TeamGroup {
  id: string;
  name: string;
  description: string;
  gender: string;

  teams: Team[];
}